src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -89,7 +90,7 @@
 
 def generic_sql_error_handler(
     err: ProgrammingError, role: Optional[str] = None, warehouse: Optional[str] = None
-):
+) -> NoReturn:
```

Comment:
nit: doesn't the rest of the code use -> None for this?

Reply:
`NoReturn` means it never returns (i.e. always raises an exception). This helps the type checker know that any code after a call to this function is unreachable